services/cluster-agent/cluster_agent/run.py

```
@@ -0,0 +1,191 @@
+    instead of propagating. Any other exception is logged and swallowed too
+    (matching services/agent's per-iteration isolation) so one bad tick
+    never kills the loop."""
+    try:
+        await fn()
+        backoff.reset()
+    except _CONNECTIVITY_ERRORS as e:
+        delay = backoff.next_delay()
+        logger.warning("%s: connectivity error (%s), retrying in %.0fs", name, e, delay)
+        await asyncio.sleep(delay)
+    except ingest_client.AuthError:
+        logger.error("%s: cluster token rejected — check the Secret matches the current token", name)
```

Ruff (0.16.3)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)